### PR TITLE
Functional Tests : Support for MariaDB

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Functional Tests : Support for MariaDB
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is :green_circle: 
| UI Tests          | 
| Fixed issue or discussion?     | N/A
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

## Linked PRs
* https://github.com/PrestaShop/QANightlyResults/pull/216
* https://github.com/PrestaShop/ga.tests.ui.pr/pull/70

## Linked Issues
* https://github.com/PrestaShop/PrestaShop/issues/36356

### CODE REVIEW

File: .github/workflows/sanity.yml [hunk 1]
```diff
@@ -20,8 +20,46 @@ jobs:
     name: Sanity
     strategy:
       matrix:
-        php: [ '8.1', '8.2', '8.3' ]
-        browser: [ 'chromium', 'firefox', 'webkit' ]
+        include:
```
Comment (jolelievre) [inline .github/workflows/sanity.yml:23]: Hmm, this means we're gonna have 18 jobs for sanity? Isn't it a bit too much to have this on ALL pull requests?

File: docker-compose.yml [hunk 2]
```diff
@@ -46,12 +59,12 @@ services:
...
       ADMIN_MAIL: ${ADMIN_MAIL:-demo@prestashop.com}
       ADMIN_PASSWD: ${ADMIN_PASSWD:-Correct Horse Battery Staple}
-    command: ["/tmp/wait-for-it.sh", "--timeout=60", "--strict", "mysql:3306", "--", "/tmp/docker_run_git.sh"]
+    command: /tmp/wait-for-it.sh --timeout=60 --strict ${DB_SERVER:-mysql}:3306 -- /tmp/docker_run_git.sh
     ports:
       - "8001:80"
       - "8002:443"
     depends_on:
-      - mysql
+      - "${DB_SERVICE:-mysql}"
```
Comment (jolelievre) [inline docker-compose.yml]: Nice!

File: .github/workflows/sanity.yml (hunk 1 above)
Comment (Progi1984) [reply .github/workflows/sanity.yml:23]: Nope... We had 9 jobs before, and now we have 12 jobs, validated with @PrestaShop/qa-functional team. (We avoid to use 18 jobs, but we test the behavior with each version of PHP).

File: docker-compose.yml (hunk 2 above)
Comment (Progi1984) [reply docker-compose.yml]: <3

**Review by SD1982 — DISMISSED** [review]

File: .github/workflows/sanity.yml (hunk 1 above)
Comment (jolelievre) [reply .github/workflows/sanity.yml:23]: Ah ok mariadb is only tested with chrome, nice!

File: docker-compose.yml [hunk 3]
```diff
@@ -49,12 +62,12 @@ services:
...
       BLACKFIRE_SERVER_ID: ${BLACKFIRE_SERVER_ID:-0}
       BLACKFIRE_SERVER_TOKEN: ${BLACKFIRE_SERVER_TOKEN:-0}
-    command: ["/tmp/wait-for-it.sh", "--timeout=60", "--strict", "mysql:3306", "--", "/tmp/docker_run_git.sh"]
+    command: /tmp/wait-for-it.sh --timeout=60 --strict ${DB_SERVER:-mysql}:3306 -- /tmp/docker_run_git.sh
     ports:
       - "8001:80"
       - "8002:443"
     depends_on:
-      - mysql
+      - "${DB_SERVICE:-mysql}"
```
Comment (jolelievre) [inline docker-compose.yml]: Why the need for two variables? `DB_SERVER` and `DB_SERVICE` which seem to always have the same value
Couldn't it be just one variable?

File: .github/actions/setup-env/action.yml [hunk 4]
```diff
@@ -82,5 +88,5 @@ runs:
         ADMIN_PASSWD: 'Correct Horse Battery Staple'
       run: |
         USER_ID=$(id -u) GROUP_ID=$(id -g) \
-        docker compose -f docker-compose.yml up -d --build
+        docker compose -f docker-compose.yml up -d prestashop-git ${{ inputs.DATABASE }} --build
```
Comment (jolelievre) [inline .github/actions/setup-env/action.yml]: Isn't the env variable enough to automatically build the proper DB service?

File: docker-compose.yml (hunk 3 above)
Comment (Progi1984) [reply docker-compose.yml]: Correct. Fixed

File: .github/actions/setup-env/action.yml (hunk 4 above)
Comment (Progi1984) [reply .github/actions/setup-env/action.yml]: I try it.

**Review by matthieu-rolland — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Review by boherm — DISMISSED** [review]

**Review by boherm — APPROVED** [review]

**Review by jolelievre — APPROVED** [review]
Thanks @Progi1984

**Comment by grooverdan** [comment]
Thank you very much @Progi1984

**Pull request merged**